frontends/parsers/p4/p4parser.ypp

```diff
@@ -1161,6 +1173,54 @@ assignmentOrMethodCallStatement
                                                  $1, $3, $6);
           $$ = new IR::MethodCallStatement(@1 + @7, mc); }
 
+    | lvalue "+=" expression ";"
```

Comment:
I am not sure this is the right semantics. This implementation will repeat the side effects on the lhs twice.

Reply:
Note most lvalue by definition could not have a side effect.
The only one which I can think of is: `headerstack[f(a)].b`  where f is defined to take 1 arguments which is inout (you could do something similar with externs too).

Reply:
regardless of the inout arguments, a method call may have arbitrary side effects within in, and the way this code duplicates the reference to the lhs, a method call inside an index would end up being called twice.  We have a pass (DoSimplifyExpressions) that exists to deal with this kind of program, but there are a number of other passes that currently must run before it (such as typeChecking) that would need to understand what is going on.  So this almost certainly needs a new IR node type (some kind of op-assignment statement) to represent this until SimplifyExpressions can turn it into a normal AssignmentStatement, and all the passes that run before need to know about it.